app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/attachment/AttachmentOptions.kt

```diff
@@ -184,33 +192,41 @@ private fun GalleryFlow(onFilePicked: (Uri) -> Unit): UseStorageRequestFlow {
 }
 
 @Composable
-private fun TakePictureFlow(tempCachePath: Path, onPictureTaken: (Uri) -> Unit): UseCameraRequestFlow {
-    val context = LocalContext.current
-    val imageAttachmentUri = context.getTempWritableImageUri(tempCachePath)
-    return rememberTakePictureFlow(
-        onPictureTaken = { hasTakenPicture ->
-            if (hasTakenPicture) {
-                onPictureTaken(imageAttachmentUri)
-            }
-        },
-        targetPictureFileUri = imageAttachmentUri,
-        onPermissionDenied = { /* TODO: Implement denied permission rationale */ }
-    )
+private fun TakePictureFlow(
+    tempWritableVideoUri: Uri?,
+    onPictureTaken: (Uri) -> Unit
+): UseCameraRequestFlow? {
+    tempWritableVideoUri?.let {
+        return rememberTakePictureFlow(
+            onPictureTaken = { hasTakenPicture ->
+                if (hasTakenPicture) {
+                    onPictureTaken(it)
+                }
+            },
+            targetPictureFileUri = it,
+            onPermissionDenied = { /* TODO: Implement denied permission rationale */ }
+        )
+    }
+    return null
 }
 
 @Composable
-private fun CaptureVideoFlow(tempCachePath: Path, onVideoCaptured: (Uri) -> Unit): UseCameraRequestFlow {
-    val context = LocalContext.current
-    val videoAttachmentUri = context.getTempWritableVideoUri(tempCachePath)
-    return rememberCaptureVideoFlow(
-        onVideoRecorded = { hasCapturedVideo ->
-            if (hasCapturedVideo) {
-                onVideoCaptured(videoAttachmentUri)
-            }
-        },
-        targetVideoFileUri = videoAttachmentUri,
-        onPermissionDenied = { /* TODO: Implement denied permission rationale */ }
-    )
+private fun CaptureVideoFlow(
+    onVideoCaptured: (Uri) -> Unit,
+    tempWritableVideoUri: Uri?
```

Comment:
Shouldn't we move the `onVideoCaptured` parameter to the second position/end so that we favor the dynamic inlining of the lambda when calling the method?